client/app/services/parameters/DateParameter.js

```diff
@@ -0,0 +1,95 @@
+import { findKey, startsWith, has, includes, isNull, values } from 'lodash';
+import moment from 'moment';
+import PropTypes from 'prop-types';
+import { Parameter } from '.';
+
+const DATETIME_FORMATS = {
+  // eslint-disable-next-line quote-props
+  'date': 'YYYY-MM-DD',
+  'datetime-local': 'YYYY-MM-DD HH:mm',
+  'datetime-with-seconds': 'YYYY-MM-DD HH:mm:ss',
+};
+
+const DYNAMIC_PREFIX = 'd_';
+
+const DYNAMIC_DATES = {
+  now: {
+    name: 'Today/Now',
+    value: () => moment(),
```

Comment:
Just curious: shouldn't it be `moment.utc()`?

Reply:
In this case we can and want to use the user's TZ (I believe it's less confusing and it reaches more use cases).

Reply:
When user selects a dynamic value for parameter - what do we pass to a server when query is executed: "magic" constant or computed value?

Reply:
Computed value (actual "dates/date ranges")

Reply:
Okay. I think, it may be an issue. Imagine that there is a query like `SELECT * FROM some_table WHERE created_at >= '{{ param }}'`, and that DB server is configured to use UTC, so `created_at` column contains UTC values. Now, imagine that I want to pass `yesterday` as param value. My local time zone is `GMT+3`, and if I'll try to use that param with `yesterday`, server will return results not for last 24 hours, but for last 27 hours - because client will pass my local time, but server will treat it as UTC - so there will be 3 hours difference. Also, this example can be extrapolated for a case with few users in different time zones - then each of them will get results for different period.

So shouldn't we always pass date/time/range params in UTC to make them predictable and not dependent on DB config and user's locale settings?

Reply:
:thinking: you have a point, from my experience I've always used `CONVERT_TZ` in the queries so it would be less confusing for the people that were only going to view it and get the results. Though everyone had the same TZ in there. 

I still have the feeling that the user TZ is the less confusing when considering the user, so I believe it's not a bug, it's one acceptable behavior for this. But we could eventually offer more options in there. 

This is certainly not in the scope of this PR, so I say we should move this discussion to somewhere else with more visibility (forum or create an issue).

Reply:
I see your point about `CONVERT_TZ`, but I described a bit different case: when we send param value to DB server - it does not know in which TZ that value actually is, and what's the difference between server's TZ and provided value. Even more: values that come from different users may be in different TZ, which is totally impossible to handle in a single query. So actually there are two solutions: 1) pass TZ in param value (ISO8601 allows this), or 2) pass UTC (we can show value in any TZ in UI, but when executing query - convert it to UTC).

It's definitely not in the scope of this PR, but we have to discuss it somewhere.

Reply:
Let's have it as `moment.utc()` in this PR (I assume this is the current behavior?) and start a discussion about how to handle timezones apart from this PR. It's a broad enough topic that requires deeper discussion.

Reply:
The current behaviour is `moment()`, so I'm assuming that's the one you meant we should keep in this scope.